climada/hazard/centroids/centr.py

```diff
@@ -172,6 +177,40 @@ def from_base_grid(land=False, res_as=360, base_file=None):
             centroids = centroids.select(reg_id=land_reg_ids)
         return centroids
 
+    @staticmethod
+    def from_geodataframe(gdf):
+        """Create centroids instance from GeoDataFrame. All columns beside the
+        geometry (or geom) column will be translated using pd.Series.to_numpy,
+        so the Series dtype will be respected.
+
+        >>> gdf = geopandas.read_file('centroids.shp')
+        >>> gdf.region_id = gdf.region_id.astype(int)  # type coercion
+        >>> centroids = Centroids.from_geodataframe(gdf)
+
+        Parameters:
+            gdf (GeoDataFrame): Where the geometry column needs to consist of
+                point features.
+        """
+        centroids = Centroids()
+
+        centroids.geometry = gdf.geometry
+        centroids.lat = gdf.geometry.y.to_numpy()
+        centroids.lon = gdf.geometry.x.to_numpy()
+
+        for col in gdf.columns:
+            if 'geom' in col:  # matches both 'geom' and 'geometry' columns
```

Comment:
what's the point in not including any additional column that contains 'geom' in its name?

Reply:
`geom` and `geometry` are synonymous in the "simple features" world of OGR; it's a convention. If you read a GDF from postgis for example, the geometry column is going to be called `geom`.

Reply:
well - then that should be reflected in line 196 `centroids.geometry = gdf.geom`
?

Reply:
No, because a `geom` column that is set to be the GDF's geometry column can also be accessed through the `.geometry` property.

Reply:
I see. Then add some inline comment. And consider `col in ['geometry', 'geom']` instead of `'geom' in col`.

Or, since explicit is better then implicit (https://www.python.org/dev/peps/pep-0020/), and since the list of such names may be even bigger, add the name of the geometry column as an optional argument:
 `def from_geodataframe(gdf, geometry_col='geometry')`.

Reply:
Thanks, can you have one last look and then hopefully merge?